src/main/kotlin/no/nav/tilleggsstonader/kontrakter/felles/Periode.kt

```diff
@@ -149,10 +149,13 @@ fun <P : Periode<LocalDate>, VAL> P.splitPerLøpendeMåneder(medNyPeriode: (fom:
  */
 fun LocalDate.sisteDagenILøpendeMåned(): LocalDate {
     val nesteMåned = this.plusMonths(1)
-    return if (this.dayOfMonth >= nesteMåned.lengthOfMonth()) {
-        nesteMåned.tilSisteDagIMåneden()
-    } else {
+    val nyttDatoErSisteDagINesteMåned by lazy {
+        this.month != nesteMåned.month && nesteMåned.dayOfMonth == nesteMåned.lengthOfMonth()
+    }
```

Comment:
Kult løst med `lazy` - det har jeg ikke sett før.

Sikkert greit å tenke over hva som er mest viktig: litt ytelses gain vs litt vanskligere kode. Begge er ok for meg så lenge det er et bevist valg ☺️